### PR TITLE
Remove parallelism in NuGet restore to workaround issue

### PR DESCRIPTION
**Pull request by joelverhagen**

### CODE REVIEW

File: scripts/dotnet-cli-build/TestTargets.cs [hunk 1]
```diff
@@ -152,7 +152,7 @@ public static BuildTargetResult RestoreTests(BuildTargetContext c)
             CleanBinObj(c, Path.Combine(c.BuildContext.BuildDirectory, "test"));
 
             CleanNuGetTempCache();
-            DotNetCli.Stage2.Restore("--fallbacksource", Dirs.TestPackages)
+            DotNetCli.Stage2.Restore("--verbosity", "verbose", "--fallbacksource", Dirs.TestPackages, "--no-cache")
```
Comment (eerhardt) [inline scripts/dotnet-cli-build/TestTargets.cs]: Why does this one get --no-cache, but the others don't?

File: scripts/dotnet-cli-build/TestTargets.cs [hunk 2]
```diff
@@ -74,17 +74,17 @@ public static BuildTargetResult RestoreTestAssetProjects(BuildTargetContext c)
             CleanNuGetTempCache();
 
             var dotnet = DotNetCli.Stage2;
-
-            dotnet.Restore("--fallbacksource", Dirs.TestPackages)
+
```
Comment (eerhardt) [inline scripts/dotnet-cli-build/TestTargets.cs]: (nit) unnecessary whitespace

File: scripts/dotnet-cli-build/TestTargets.cs (hunk 1 above)
Comment (joelverhagen) [inline scripts/dotnet-cli-build/TestTargets.cs]: This is a mistake. Will fix.

File: scripts/dotnet-cli-build/TestTargets.cs (hunk 2 above)
Comment (joelverhagen) [inline scripts/dotnet-cli-build/TestTargets.cs]: Fixed.

File: scripts/dotnet-cli-build/TestTargets.cs (hunk 1 above)
Comment (joelverhagen) [inline scripts/dotnet-cli-build/TestTargets.cs]: Fixed.

**Comment by anurse** [comment]
Sounds like we should just take this to unblock OSX builds. The RHEL and Windows_2016 failures look unrelated. /cc @piotrpMSFT @davidfowl 

We do need a rebase though, @joelverhagen

**Comment by joelverhagen** [comment]
@anurse, yup, I will do that right before merging -- is that okay? That's the process we follow in NuGet.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Windows_2016 x86 Debug Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Windows_2016 x86 Debug Build please.

**Comment by joelverhagen** [comment]
Rebased.

**Comment by anurse** [comment]
@joelverhagen Yeah. We usually use the Merge button here so once you get green, just click that guy :)
